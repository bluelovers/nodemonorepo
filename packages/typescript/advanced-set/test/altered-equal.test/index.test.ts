import { AlteredEqual } from '../../index'

it('calls injected function', () => {
  const track = Array<[typeof act, number, number]>()
  let act: 'has' | 'add' | 'delete'

  const equal = (a: number, b: number): boolean => {
    track.push([act, a, b])
    return a === b
  }

  const set = new AlteredEqual<number>(Set, equal)

  act = 'add'
  set.add(0)
  set.add(1)

  act = 'has'
  ; [0, 1, 2, 3].map(x => set.has(x))

  act = 'delete'
  ; [0, 1, 2, 3].map(x => set.delete(x))

  expect(track).not.toEqual([])
  expect(track).toMatchSnapshot()
})

it('add() works as intended', () => {
  const element = Symbol()
  const set = new AlteredEqual(Set)

  const before = set.has(element)
  set.add(element)
  const after = set.has(element)

  expect({
    before,
    after
  }).toEqual({
    before: false,
    after: true
  })
})

it('has() works as intended', () => {
  const set = new AlteredEqual(Set)
  set.add(0)
  set.add(1)

  const expected = [true, true, false, false]
  const received = [0, 1, 2, 3].map(x => set.has(x))
  expect(received).toEqual(expected)
})

it('delete() works as intended', () => {
  const element = Symbol()
  const set = new AlteredEqual(Set)

  const before = set.has(element)
  set.add(element)
  const between = set.has(element)
  set.delete(element)
  const after = set.has(element)

  expect({
    before,
    between,
    after
  }).toEqual({
    before: false,
    between: true,
    after: false
  })
})
