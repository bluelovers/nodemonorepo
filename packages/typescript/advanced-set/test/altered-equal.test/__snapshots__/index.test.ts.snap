// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls injected function 1`] = `
Array [
  Array [
    "add",
    1,
    0,
  ],
  Array [
    "has",
    0,
    0,
  ],
  Array [
    "has",
    1,
    0,
  ],
  Array [
    "has",
    1,
    1,
  ],
  Array [
    "has",
    2,
    0,
  ],
  Array [
    "has",
    2,
    1,
  ],
  Array [
    "has",
    3,
    0,
  ],
  Array [
    "has",
    3,
    1,
  ],
  Array [
    "delete",
    0,
    0,
  ],
  Array [
    "delete",
    1,
    1,
  ],
]
`;
